### ISSUE TITLE
Many messages are being output to the stderr when starting from a terminal

### ISSUE THREAD

**Issue by shlomif** [equalsraf/neovim-qt#509]
When I call the latest nvim-qt on konsole / tmux / plasma 5 / mageia v7 I get many debug messages in the konsole. Like:

```
Received unknown redraw notification "update_menu" ()
Received unknown redraw notification "update_menu" ()
Received unknown redraw notification "update_menu" ()
Received unknown redraw notification "update_menu" ()
Received unknown redraw notification "update_menu" ()
Received unknown redraw notification "flush" ()
Received unknown redraw notification "update_menu" ()
Received unknown redraw notification "update_menu" ()
Received unknown redraw notification "update_menu" ()
Received unknown redraw notification "update_menu" ()
Received unknown redraw notification "update_menu" ()
Received unknown redraw notification "update_menu" ()
Received unknown redraw notification "update_menu" ()
Received unknown redraw notification "update_menu" ()
Received unknown redraw notification "flush" ()
Received unknown redraw notification "flush" ()
Received unknown redraw notification "flush" ()
Received unknown redraw notification "flush" ()
Received unknown redraw notification "flush" ()
Received unknown redraw notification "flush" ()
Received unknown redraw notification "flush" ()
Received unknown redraw notification "flush" ()
Received unknown redraw notification "flush" ()
Received unknown redraw notification "flush" ()
Received unknown redraw notification "flush" ()

```

gvim OTOH is very silent.

**Comment by gagbo** [comment]
Duplicate of #507 

tl;dr : these are harmless debug messages, enabled because of a combination of compile flags, and environment options regarding the handling of [QtDebugMsg](http://doc.qt.io/qt-5/qtglobal.html#QtMsgType-enum).

I use [kdebugsettings](https://github.com/KDE/kdebugsettings) to set/unset debug messages in runtime using a GUI, but maybe at this point we could also look at solutions to prevent part of this. I don't see an easy solution to this (except using kdebugsettings) because I suspect build in packaged versions of nvim-qt include some sort of debug flags for automatic bug reporting tools ; I might be wrong though

**Comment by shlomif** [comment]
> Duplicate of #507
> 
> tl;dr : these are harmless debug messages, enabled because of a combination of compile flags, and environment options regarding the handling of [QtDebugMsg](http://doc.qt.io/qt-5/qtglobal.html#QtMsgType-enum).
> 
> I use [kdebugsettings](https://github.com/KDE/kdebugsettings) to set/unset debug messages in runtime using a GUI, but maybe at this point we could also look at solutions to prevent part of this. I don't see an easy solution to this (except using kdebugsettings) because I suspect build in packaged versions of nvim-qt include some sort of debug flags for automatic bug reporting tools ; I might be wrong though
> 
> Also, `2>/dev/null` should discard all those messages

Thanks!

Building a Release build made them go away. With command line use these messages are not entirely harmless because they make the existing prompt disappear which is annoying. How can I use kdebugsettings to do that? I do not see "nvim" or "neovim" there.

**Comment by gagbo** [comment]
Loose translation from french here. In kdebugsettings, under "Custom rules" I add a "*.debug=true" or "*.debug=false" depending on if I want the debug output or not. Usually using a wildcard is not a good pattern, but as far as I understand, nvim-qt debug messages do not register under any "category" I could use there

**Comment by equalsraf** [comment]
These should not show up in a release build. If you are building from source make sure to set the build type like this

```
cmake -DCMAKE_BUILD_TYPE=Release ..
```

If this was installed from a package, then the package needs to be fixed accordingly.
